src/main.rs

```diff
@@ -508,7 +508,7 @@ impl Context {
 
             // Construct the path that contains the documentation inside the tarball.
             let tarball_dir = format!("{tarball_prefix}/rustc-docs/share/doc/rust/html");
-            let tarball_dir_new = format!("{tarball_dir}/rustc");
+            let tarball_dir_new = format!("{tarball_dir}/rustc-docs");
```

Comment:
We generally want tip of promote release to work for all channels. This particular component probably doesn't matter too much but I'd rather write the code in a forwards and backwards compatible way... can we do something to support both the old and new way? Maybe check if either directory exists? Or include an empty flag file in the component that we can test for here?